data-generation/src/main/scala/thesis/SnapshotDelta.scala

```diff
@@ -165,7 +168,8 @@ object SnapshotDeltaObject {
   }
 
   def applyEdgeLogsToSnapshot(snapshot: Graph[LTSV.Attributes, LTSV.Attributes], logs: RDD[LogTSV]): RDD[Edge[Attributes]] = {
-    val previousSnapshotEdgesKV = snapshot.edges.map(edge => ((edge.dstId, edge.srcId), edge.attr))
+    val uuid = java.util.UUID.randomUUID.toString.take(5)
+    val previousSnapshotEdgesKV = snapshot.edges.map(edge => ((edge.srcId, edge.dstId), edge.attr))
```

Comment:
this line is the fix. (I was supposed to have a single commit for this but I messed up)